qt_ui/widgets/QTopPanel.py

```diff
@@ -127,15 +138,21 @@ def setGame(self, game: Optional[Game]):
         self.budgetBox.setGame(game)
         self.factionsInfos.setGame(game)
 
-        self.passTurnButton.setEnabled(True)
-        if game and game.turn > 0:
-            self.passTurnButton.setText("Pass Turn")
+        state = game.check_win_loss()
+        enabled = state == TurnState.CONTINUE
+        for controller in self.controls:
+            controller.setEnabled(enabled)
 
-        if game and game.turn == 0:
+        if game.turn > 0:
+            self.passTurnButton.setText("Pass Turn")
+        elif game.turn == 0:
             self.passTurnButton.setText("Begin Campaign")
             self.proceedButton.setEnabled(False)
+            self.simSpeedControls.setEnabled(False)
         else:
-            self.proceedButton.setEnabled(True)
+            assert (
+                game.turn >= 0
+            ), f"FUBAR: game.turn out of bounds!\n  value = {game.turn}"
```

Comment:
Raise an `AssertionError ` if `game.turn` < 0, because this should really NEVER happen, so if it does, we'd like to know...

Reply:
Yes we would, so raise `RuntimeError` rather than using `assert`, which is disabled in release builds. I think the only places `assert` is useful is for forcing the type system to acknowledge an otherwise inexpressible constraint, and tests.